lib/Common/Memory/Recycler.cpp

```diff
@@ -5674,8 +5731,57 @@ Recycler::FinishConcurrentCollect(CollectionFlags flags)
 #endif
         GCETW_INTERNAL(GC_STOP, (this, ETWEvent_ConcurrentRescan));
     }
+#if ENABLE_CONCURRENT_GC && ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP
+    else if (collectionState == CollectionStateConcurrentSweepPass1Wait)
+    {
+        this->FinishSweepPrep();
+        this->collectionState = CollectionStateConcurrentSweepPass2;
```

Comment:
I'd move this into the if. And I'd change the else condition to call StartConcurrent instead of SetEvent